### PR TITLE
Remove container type from kubelet runtime labels

### PR DESCRIPTION
**Pull request by verb**
We've changed the Ephemeral Containers API, and container type will no
longer be required. Since this is the only feature using it, remove it.

This reverts commit ba6f31a6c60e9bead6947c01e3cddfa6148523c5.

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**: This cleans up container type metadata in the kubelet which is no longer needed. This would be tracked instead in the apiserver

**Which issue(s) this PR fixes**:

WIP #27140

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**: 

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubelet/container/runtime.go [hunk 1]
```diff
@@ -254,13 +254,6 @@ const (
 	ContainerStateUnknown ContainerState = "unknown"
 )
 
-type ContainerType string
-
-const (
-	ContainerTypeInit    ContainerType = "INIT"
-	ContainerTypeRegular ContainerType = "REGULAR"
-)
-
```
[the remark this answers is not in the document]
Comment (krmayankk) [reply pkg/kubelet/container/runtime.go]: @verb ^^

[the remark this answers is not in the document]
Comment (verb) [reply pkg/kubelet/container/runtime.go]: The original design of Debug Containers required the kubelet to track this information, but the [current proposal](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/node/troubleshoot-running-pods.md) has changes to the core API that make this no longer necessary.

This was hidden behind a feature flag, so it wasn't used anywhere.

**Pull request merged**